tests/ui/collapsible_str_replace.rs

```diff
@@ -0,0 +1,68 @@
+// run-rustfix
+
+#![warn(clippy::collapsible_str_replace)]
+
+fn get_filter() -> &'static str {
+    "u"
+}
+
+fn main() {
+    let misspelled = "hesuo worpd";
+
+    let p = 'p';
+    let s = 's';
+    let u = 'u';
+    let l = "l";
+
+    // LINT CASES
+    let _ = misspelled.replace('s', "l").replace('u', "l");
+
+    let _ = misspelled.replace('s', l).replace('u', l);
+
+    let _ = misspelled.replace('s', "l").replace('u', "l").replace('p', "l");
+
+    let _ = misspelled
+        .replace('s', "l")
+        .replace('u', "l")
+        .replace('p', "l")
+        .replace('d', "l");
+
+    let _ = misspelled.replace(s, "l").replace('u', "l");
+
+    let _ = misspelled.replace(s, "l").replace('u', "l").replace('p', "l");
+
+    let _ = misspelled.replace(s, "l").replace(u, "l").replace('p', "l");
+
+    let _ = misspelled.replace(s, "l").replace(u, "l").replace(p, "l");
+
+    let _ = misspelled.replace('s', "l").replace('u', "l").replace('p', "x");
+
+    let _ = misspelled.replace('s', "x").replace('u', "l").replace('p', "l");
+
+    // NO LINT CASES
+    let _ = misspelled.replace('s', "l");
+
+    let _ = misspelled.replace(s, "l");
+
+    let _ = misspelled.replace('s', "l").replace('u', "p");
+
+    let _ = misspelled.replace(&get_filter(), "l");
+
+    let _ = misspelled.replace(&['s', 'u', 'p'], "l");
+
+    let _ = misspelled.replace(&['s', 'u', 'p'], l);
+
+    let _ = misspelled.replace(&['s', 'u'], "l").replace(&['u', 'p'], "l");
+
+    let _ = misspelled.replace('s', "l").replace(&['u', 'p'], "l");
+
+    let _ = misspelled.replace(&['s', 'u'], "l").replace('p', "l");
+
+    let _ = misspelled.replace(&['s', u, 'p'], "l");
+
+    let _ = misspelled.replace(&[s, u, 'p'], "l");
+
+    let _ = misspelled.replace(&[s, u, p], "l");
+
+    let _ = misspelled.replace(&[s, u], "l").replace(&[u, p], "l");
```

Comment:
Please add a comment here, that this could get linted in the future and combined to `[s, u, p]`.